Refactor experiment page bundle

lingyun1010 opened the pull request:
We are going to simplify the logic and deployment of experiment page router component and trying to move the bundle to [the front end component repository](https://github.com/ebi-gene-expression-group/atlas-components/tree/chore%2Frefactor-experiment-page-router-component). The package has not been officially published yet, so in this PR, I only installed a local compiled package, you may test the refactoring results by doing the same.

lingyun1010 committed:
Delete all the src and html files from experiment-

page bundle

lingyun1010 committed:
Apply new published pacakge for experiment page

lingyun1010 committed:
Refactor package.json with experiment page package

lingyun1010 committed:
Install a published npm package of experiment page

ke4 commented on app/src/main/javascript/bundles/experiment-page/package.json:
```diff
@@ -1,43 +1,19 @@
-    "webpack-cli": "^3.3.12",
-    "webpack-dev-server": "^3.11.0",
-    "whatwg-fetch": "^3.4.1"
+    "@ebi-gene-expression-group/scxa-experiment-page": "^1.0.0-batch",
```
Is the version not `1.0.0-test`?
that's what I saw in your other PR: https://github.com/ebi-gene-expression-group/atlas-components/pull/182/files in the `package.json` file.

upendrakumbham replied:
yes, it is. Now it is` -batch` here.

lingyun1010 replied:
Yes, it is `1.0.0-batch` . I will push the front end component version commit.
```A new version of the package @ebi-gene-expression-group/scxa-experiment-page (1.0.0-batch) was published at 2024-11-26T16:06:01.205Z from 193.62.194.247.```

ke4 replied:
Yes, it is now, but it was not when I commented it.

ke4 commented on app/src/main/javascript/bundles/experiment-page/index.js:
```diff
@@ -1,20 +1,21 @@
+        accessKey={options.content.accessKey}
+        species={options.content.species}
+        tabs={options.content.tabs}
+        anatomogram={options.content.anatomogram}
```
I can not see `anatomogram` property in the `ExperimentPageRouter` component in the `scxa-experiment-page` package.

lingyun1010 replied:
I don't understand your questions. What do you mean by `property`? Anatomogram is one of the tabs and if you run the webpack server in the `atlas-component`, you may see the content of anatomogram.

ke4 replied:
In this file: [packages/scxa-experiment-page/src/ExperimentPageRouter.js](https://github.com/ebi-gene-expression-group/atlas-components/blob/5b717676e8e96077b4fbda44f25006fd794dfbac/packages/scxa-experiment-page/src/ExperimentPageRouter.js) I can see all of these properties defined:

```
const TabCommonPropTypes = {
  atlasUrl: PropTypes.string.isRequired,
  experimentAccession: PropTypes.string.isRequired,
  species: PropTypes.string.isRequired,
  accessKey: PropTypes.string,
  resourcesUrl: PropTypes.string
}
...
ExperimentPageRouter.propTypes = {
  ...TabCommonPropTypes,
  tabs: PropTypes.arrayOf(PropTypes.shape({
    type: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    props: PropTypes.object.isRequired
  })).isRequired
}
```

but not `anatomogram`.
Could you tell me whether `ExperimentPageRouter` is using this property?
If not, then why are we setting it?
Thanks

ke4 reviewed (changes requested):
Thanks for your work.
I added just 2 comments to double check my understanding of the related code in the scxa-experiment-page package.

upendrakumbham reviewed (changes requested):
Please add validations where you need before fetching data and consider `isRequired `props validations mandatory please.

lingyun1010 commented:
> Please add validations where you need before fetching data and consider `isRequired `props validations mandatory please.

I think that's not this PR about. That is one of your previous work isn't it? The validation?

lingyun1010 committed:
Merge branch 'develop' into chore/refactor-experiment-page-bundle

lingyun1010 committed:
Update package-lock

ke4 reviewed (changes requested):
Hi,
I am still not sure what is the `anatomogram` property in the `ExperimentPageRouter`.
I added a longer comment regarding to that one.
Could you please clarify it?
Thanks

lingyun1010 committed:
Merge branch 'chore/refactor-experiment-page-bundle' of https://github.com/ebi-gene-expression-group/atlas-web-single-cell into chore/refactor-experiment-page-bundle

lingyun1010 committed:
Remove redundant props for experiment page router

lingyun1010 commented:
> Hi, I am still not sure what is the `anatomogram` property in the `ExperimentPageRouter`. I added a longer comment regarding to that one. Could you please clarify it? Thanks

Yes, I think you are right, we have `anatomogram` props under variable `tabs`, i.e., `anatomogram` is an entry of `tabs`, but here is a redundant or bug line, as we don't have `anatomogram` props under `content`. I removed this line, please take a further look.

ke4 reviewed (dismissed):
LGTM
Thanks for clarifying my suggestion.

upendrakumbham reviewed (dismissed):
LGTM!

lingyun1010 committed:
Keep npm packages up to date

lingyun1010 committed:
Install the latest version of scxa-experiment-page
